galter_subjects_utils/contrib/mesh/converter.py

```diff
@@ -143,46 +143,60 @@ def _analyze_dst(self):
 
     def _analyze_replace(self):
         """Determine which subjects have been replaced."""
-        for label, analysis in self._label_to_analysis.items():
-            if analysis.seen:
-                continue
 
-            # get new label
-            # direct replacement
+        def find_replacement_label(label):
+            """Return replacement label or None."""
             new_label = self.replacements.get(label)
-            if not new_label:
-                # maybe label is qualified, so check if root is replaced
-                root_label, slash, qualifier = label.rpartition("/")
-                new_root_label = self.replacements.get(root_label)
-                if new_root_label:
-                    new_label = new_root_label + slash + qualifier
-                else:
-                    new_label = None
-
-            if not new_label:
-                # No replacement
-                continue
+            if new_label:
+                return new_label
+
+            # maybe label is qualified, so check if root is replaced
+            root_label, slash, qualifier = label.rpartition("/")
+            new_root_label = self.replacements.get(root_label)
+            if not new_root_label:
+                return None
+            new_label = new_root_label + slash + qualifier
+            return new_label
+
+        def find_replacement_id(label):
+            """Return replacement id or None if not replaced."""
+            replacement_label = find_replacement_label(label)
+            if not replacement_label:
+                return None
+
+            # is replacement_label for existing and kept subject?
+            replacement_analysis = self._label_to_analysis.get(replacement_label)  # noqa
+            if replacement_analysis and replacement_analysis.seen:
+                return replacement_analysis.id
+
+            # is replacement_label for newly added subject?
+            replacement_subject = next(
```

Comment:
The `next` function always takes me a second to grasp. It's pythonic, I'm just not used to it still.

Reply:
Yes, it's a somewhat round-about way of implementing "find first instance in array satisyfing these conditions". It's more compact than a for-loop for that purpose and more efficient.